pentaho/src/main/java/pt/webdetails/cda/CdaUtils.java

```diff
@@ -469,13 +474,32 @@ public String clearCache( @Context HttpServletResponse servletResponse,
     return msg;
   }
 
+  public IUserSession getUserSession() {
+    return new PentahoSessionUtils().getCurrentSession();
+  }
+
+  public boolean canCreateContent() {
+    IAuthorizationPolicy authorizationPolicy = SecurityHelper.getInstance().getAuthorizationPolicy();
+    if ( authorizationPolicy == null ) {
+      authorizationPolicy = PentahoSystem.get( IAuthorizationPolicy.class );
+      if ( authorizationPolicy == null ) {
+        logger.warn( "Couldn't retrieve Authorization Policy" );
+        return getUserSession().isAdministrator();
+      }
+    }
+    return authorizationPolicy.isAllowed( RepositoryCreateAction.NAME );
```

Comment:
Is this the only way to get to the roles/permissions of the user?

I seem to remember we got there some other way. @afrjorge can you take a look?

Reply:
@pamval I used this code https://github.com/webdetails/cde/blob/feeef2e2c52ab9339df8cb6840973bc7e3dbd83d/cde-pentaho5/src/pt/webdetails/cdf/dd/PentahoCdeEnvironment.java#L218-L228 as example.
So if we are doing it in this way in cde, should we change it in cda?

Reply:
The code in CDE does not need a new method in the ISecurityHelper interface. This one does. My question is why do we need to introduce a new method to that interface?

Reply:
Well, there is another working solution which is not causing the ISecurityHelper interface modification: https://github.com/Advent51/cda/commit/135a0e2f4e6253d8c7936c695f812291d5e6f288
But we need to add two more methods in ICdaEnvironment interface.
@pamval please review, may be the new solution is more appropriate.

Reply:
yeah, I like that one better